commit/chainfee/outcome.go

```diff
@@ -0,0 +1,185 @@
+		p.lggr,
+		"FeeComponents",
+		aggObs.FeeComponents,
+		twoFChainPlus1,
+		// Aggregator function
+		func(vals []types.ChainFeeComponents) types.ChainFeeComponents {
+			executionFees := make([]cciptypes.BigInt, len(vals))
+			dataAvailabilityFees := make([]cciptypes.BigInt, len(vals))
+			for i, feeComp := range vals {
+				executionFees[i] = cciptypes.NewBigInt(feeComp.ExecutionFee)
+				dataAvailabilityFees[i] = cciptypes.NewBigInt(feeComp.DataAvailabilityFee)
+			}
+			return types.ChainFeeComponents{
+				ExecutionFee:        consensus.Median(executionFees, consensus.BigIntComparator).Int,
+				DataAvailabilityFee: consensus.Median(dataAvailabilityFees, consensus.BigIntComparator).Int,
+			}
+		},
+	)
+
+	nativeTokenPrices := consensus.GetConsensusMapAggregator(
+		p.lggr,
+		"NativeTokenPrices",
+		aggObs.NativeTokenPrices,
+		twoFChainPlus1,
+		// Aggregator function
+		func(vals []cciptypes.BigInt) cciptypes.BigInt {
+			return consensus.Median(vals, consensus.BigIntComparator)
+		},
+	)
+
+	consensusObs := Observation{
+		FChain:            fChains,
+		FeeComponents:     feeComponents,
+		NativeTokenPrices: nativeTokenPrices,
+		ChainFeeUpdates:   chainFeeUpdatesConsensus,
+		TimestampNow:      timestamp,
+	}
+
+	return consensusObs, nil
+}
+
+func aggregateObservations(aos []plugincommon.AttributedObservation[Observation]) AggregateObservation {
+	aggObs := AggregateObservation{
+		FeeComponents:     make(map[cciptypes.ChainSelector][]types.ChainFeeComponents),
+		NativeTokenPrices: make(map[cciptypes.ChainSelector][]cciptypes.BigInt),
+		FChain:            make(map[cciptypes.ChainSelector][]int),
+		ChainFeeUpdates:   make(map[cciptypes.ChainSelector][]Update),
+		Timestamps:        []time.Time{},
+	}
+
+	for _, ao := range aos {
+		obs := ao.Observation
+
+		// FeeComponents
+		for chainSel, feeComp := range obs.FeeComponents {
+			aggObs.FeeComponents[chainSel] = append(aggObs.FeeComponents[chainSel], feeComp)
+		}
+
+		// NativeTokenPrices
+		for chainSel, tokenPrice := range obs.NativeTokenPrices {
+			aggObs.NativeTokenPrices[chainSel] = append(aggObs.NativeTokenPrices[chainSel], tokenPrice)
+		}
+
+		// FChain
+		for chainSel, f := range obs.FChain {
+			aggObs.FChain[chainSel] = append(aggObs.FChain[chainSel], f)
+		}
+
+		for chainSel, feeUpdate := range obs.ChainFeeUpdates {
+			aggObs.ChainFeeUpdates[chainSel] = append(aggObs.ChainFeeUpdates[chainSel], feeUpdate)
+		}
+
+		// Timestamps
+		aggObs.Timestamps = append(aggObs.Timestamps, obs.TimestampNow)
+	}
+
+	return aggObs
+}
+
+// getGasPricesToUpdate checks which chain fees need to be updated based on the observed chain fee prices and
+// the fee quoter updates.
+// A chain fee is selected for update if it meets one of 2 conditions:
+// 1. If time passed since the last update is greater than the stale threshold.
+// 2. If deviation between the fee quoter and latest observed chain fee exceeds the chain's configured threshold.
+func (p *processor) getGasPricesToUpdate(
```

Comment:
Maybe we want to rename `Gas Price` to `Chain Fee` in several places?

Reply:
The current final report is GasPrice, that's why I use it only here